aggregator/src/events/events_cache.rs

```diff
@@ -0,0 +1,143 @@
+use std::{ops::DerefMut, sync::Mutex};
+
+use crate::common::{constants::EVENT_STATS_DATA, error};
+use k8s_openapi::api::core::v1::ConfigMap;
+use mbus_api::{
+    mbus_nats::NatsMessageBus,
+    message::{Action, Category, EventMessage},
+    Bus,
+};
+use once_cell::sync::OnceCell;
+use serde::{Deserialize, Serialize};
+use snafu::ResultExt;
+static CACHE: OnceCell<Mutex<Cache>> = OnceCell::new();
+
+/// EventSet captures the type of events.
+#[derive(Serialize, Deserialize, Clone, Debug, Default)]
+pub struct EventSet {
+    pub pool: Pool,
+    pub volume: Volume,
+}
+
+impl EventSet {
+    pub fn from_event_store(init_data: ConfigMap) -> error::Result<Self> {
+        let data = init_data
+            .data
+            .ok_or(error::ReferenceConfigMapNoData.build())?;
+        let value = data.get(EVENT_STATS_DATA).ok_or(
+            error::ReferencedKeyNotPresent {
+                key: EVENT_STATS_DATA.to_string(),
+            }
+            .build(),
+        )?;
+
+        let event_set = serde_json::from_str(value)
+            .context(error::EventSerdeDeserialization { event: value })?;
+        Ok(event_set)
+    }
+
+    fn inc_counter(&mut self, category: Category, action: Action) -> error::Result<()> {
+        match category {
+            Category::Pool => self.pool.inc_counter(action),
+            Category::Volume => self.volume.inc_counter(action),
+        }
+    }
+}
+
+impl From<&mut EventSet> for EventSet {
+    fn from(event_set: &mut EventSet) -> Self {
+        EventSet {
+            pool: event_set.pool.clone(),
+            volume: event_set.volume.clone(),
+        }
+    }
+}
+
+/// Volume related events.
+#[derive(Serialize, Deserialize, Clone, Default, Debug)]
```

Comment:
Might be worth making events_cache a dir module, and have volume & pools as separate files, as it's likely we'll add more..